### DISCUSSION TITLE
How to develop ohif plugin？

### DISCUSSION THREAD

**Discussion by TheSeaOfStars123** [Project-MONAI/MONAILabel#811] [Q&A] [answered]
Hello, 

   I would like to know how to develop ohif plugin? Now  I have successfully executed `./build.sh`, I found that each request calls the compiled js files in the `monailabel/endpoints/static/ohif` folder. Is there any way to avoid recompiling `./build.sh` every time after modifying the code, can you tell me the specific steps for development, I am new to using uvicorn and fastapi.

   Thanks in advance！

**Comment by SachidanandAlle** [comment, marked as answer]
You can uncomment the remove line for viewer.. and use that directly with Orthanc independently for development..

**Reply by TheSeaOfStars123** [reply]
Do you mean to uncomment these lines?
https://github.com/Project-MONAI/MONAILabel/blob/ad2e081ae69e88cb4f3c112a071c9b5a435b04f5/plugins/ohif/build.sh#L58

Then my steps to start the project are as follows:
1. cd `plugins/ohif/Viewer` && yarn run dev:orthanc
2. start monailabel server: `monailabel start_server --app apps/radiology --studies http://127.0.0.1:8042/dicom-web`
3. Access http://127.0.0.1:3000
I'm a little unsure about the second step, are the above steps correct?

**Reply by diazandr3s** [reply]
Hi @TheSeaOfStars123,

What do you mean by developing the OHIF plugin? It is not clear to me what you'd like to do. 
Do you want to use MONAI Label on OHIF? Are you using the repo directly or are you installing monai label through the PyPI?

**Reply by TheSeaOfStars123** [reply]
Yes, I use MONAI Label on OHIF and I want to add some new functional modules. 

I think I found the solution:
https://github.com/Project-MONAI/MONAILabel/blob/ad2e081ae69e88cb4f3c112a071c9b5a435b04f5/plugins/ohif/README.md?plain=1#L22
 I will try it according to the documentation, thanks.

**Reply by SachidanandAlle** [reply]
We removed submodule.. so u cannsave the copy of viewer by commenting `rm -rf Viewers`

**Comment by adwaykanhere** [comment]
Hi,

Thanks for the answer. I was able to get the default OHIF running with the MONAI plugin. What should I do if I already have a version of OHIF locally that I'm working on and only want to integrate monailabel's plugin with it?

**Reply by diazandr3s** [reply]
Hi @adwaykanhere,

Thanks for the update.

> What should I do if I already have a version of OHIF locally that I'm working on and only want to integrate monailabel's plugin with it?

I was wondering how different is the OHIF version you have locally from the one MONAI Label has. Is there any feature that the basic one doesn't? 
I ask this as it may be easier to just use the one available in MONAI Label. 

Maybe @SachidanandAlle can comment more on this.

**Comment by SachidanandAlle** [comment]
You can build plugin with your own ohif version (if it's 2.x)

Only couple of steps to follow to include monailabel plugin as part your own ohif build..
https://github.com/Project-MONAI/MONAILabel/blob/main/plugins/ohif/build.sh#L51-L52
(don't get confused with 2 lines)

At the end you can include the https://github.com/Project-MONAI/MONAILabel/tree/main/plugins/ohif/monai-label folder along with other plugins in your OHIF repo.. and add an entry to be included into `./platform/viewer/src/index.js`

**Comment by adwaykanhere** [comment]
Thanks so much, @diazandr3s and @SachidanandAlle 

@diazandr3s I'm building OHIF with [total metabolic tumor volume ](https://ohif.org/modes/tmtv-mode)support. I'm perfectly comfortable with the version from monailabel too.

**Reply by blair49** [reply]
@adwaykanhere were you able to integrate MONAI label in your existing OHIF repo. If yes, what were the exact steps you followed? 
Also which version of OHIF and MONAI label did you use?
